internal/database/factory.go

```diff
@@ -7,12 +7,16 @@ import (
 	"github.com/pkg/errors"
 )
 
-// Database is a interface to expose database methods
-type Database interface {
+// ItemReader is a interface to expose methods to fetch items
+type ItemReader interface {
 	GetAll(ctx context.Context) ([]Item, error)
 	GetStories(ctx context.Context) ([]Item, error)
 	GetJobs(ctx context.Context) ([]Item, error)
-	Insert(ctx context.Context, item Item) error
+}
+
+// ItemWriter is a interface to expose methods to store items
+type ItemWriter interface {
+	Write(ctx context.Context, item Item) error
```

Comment:
Out of interest, why did you choose to separate the two?

Reply:
I was thinking for testing it'd be easier to mock the part we care about during a specific test rather than implement the entire db interface. I'll be honest, I still need to look properly into the mocking link you sent in a previous PR so this might be an non-issue.

Reply:
Yeah that makes sense, I think if there were lots of different methods I'd probably agree. Maybe an optimisation too soon atm

Reply:
Just remembered the original reasoning. 

The cache layer is implemented as a decorator around the database and therefore shares the same methods. By having the `Insert`/`Write` function as part of that singular interface I'd be required to add a no-op `Insert` method to the cache struct to satisfy the interface.

Reply:
Thought I was being smart :D but yeah that makes sense.